src/exberry_int/integration_exberry.py

```diff
@@ -361,22 +373,33 @@ def mass_cancel(mass_cancel_req):
         return mass_cancel_json
 
 
-
     async def connect():
-        LOG.info(f"Connecting to the Exberry Trading API at {env.tradingApiUrl} ...")
-        ws = await ClientSession().ws_connect(env.tradingApiUrl)
-        LOG.info("...Connected to the Exberry Trading API")
+        ws = None
+        tasks = []
+        try:
+            LOG.info(f"Connecting to the Exberry Trading API at {env.tradingApiUrl} ...")
+            ws = await ClientSession().ws_connect(env.tradingApiUrl)
+            LOG.info("...Connected to the Exberry Trading API")
 
-        LOG.info(f"Preparing producer coroutine...")
-        sender_task = asyncio.create_task(producer_coro(ws))
+            LOG.info(f"Preparing producer coroutine...")
+            sender_task = asyncio.create_task(producer_coro(ws))
 
-        LOG.info(f"Preparing consumer coroutine...")
-        receiver_task = asyncio.create_task(consumer_coro(ws))
+            LOG.info(f"Preparing consumer coroutine...")
+            receiver_task = asyncio.create_task(consumer_coro(ws))
 
-        LOG.info(f"Requesting market session...")
-        await request_session(env.apiKey, env.secret, ws)
+            LOG.info(f"Requesting market session...")
+            await request_session(env.apiKey, env.secret, ws)
 
-        asyncio.gather(*[sender_task, receiver_task])
+            tasks = [sender_task, receiver_task]
+            asyncio.gather(*tasks)
+        except Exception as e:
+            LOG.warn(f'connection error: {e}, cancelling tasks and cleaning up...')
+            [t.cancel() for t in tasks]
```

Comment:
Why is this a list comprehension when you're not concerned with the return value?